x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -63,16 +64,41 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
     const search = () => this.runSearch({ id, ...request }, searchOptions);
 
     this.pendingCount$.next(this.pendingCount$.getValue() + 1);
+    const isCurrentSession = () =>
+      !!options.sessionId && options.sessionId === this.deps.session.getSessionId();
+
+    const untrackSearch = isCurrentSession() && this.deps.session.trackSearch({ abort });
+
+    // track if this search's session will be send to background
+    // if yes, then we don't need to cancel this search when it is aborted
+    let isSavedToBackground = false;
+    const savedToBackgroundSub =
+      isCurrentSession() &&
+      this.deps.session.state$
+        .pipe(
+          skip(1), // ignore any state, we are only interested in transition x -> BackgroundLoading
+          filter((state) => isCurrentSession() && state === SessionState.BackgroundLoading),
```

Comment:
Same should apply if the session has finished loading.

Reply:
Why not use `sessionService.isStored` instead?

Reply:
> Why not use sessionService.isStored instead?

This is what I initially thought of, but it didn't work for me: 
sessionService tracks state of the CURRENT session and at the point when we cancel the search it might be already tracking different session. 

Let's assume we have a search that will be aborted because of navigating away. 
Consider this scenario: 

1. Session is created, search is fired
2. before we start search, `isStored` is still false 
3. Session is saved, `isStored` becomes true
4. User navigates away. session is cleared. 
5. Search gets canceled, we are in `catchError`, `sessionService.isStored` is false, because session was cleared. So we don't know if the search that was cancelled was stored or not. 

To support this use case using session service we should keep state of older sessions. 
Other simpler alternative for this specific use case was to subscribe to state change into `backgroundLoading` state and keep that bool flag in closure.  

> Same should apply if the session has finished loading.

Don't think so, it seems that transition to `BackgroundLoading` is what we really interested in, if I am not missing something.